TEST_DEV_GUIDE.md

```diff
@@ -0,0 +1,87 @@
+
+- @AcceptanceTest - Tests based on requirements specifically stated in the RFC. These are tests that will run to verify confirmance of the implemented protocol in the Agents.
+- @DerivedFunctionalTest - Tests derived on requirements but not specifically stated in the RFC.
+- @P1 - Test Priority
+- @P2 - Test Priority
+- @P3 - Test Priority
+- @P4 - Test Priority
+- @NegativeTest - Test that attempts to break the software. ie. change workflow order, use invalid data, etc.
+- @ExceptionTest - Tests that are based on requirements that suggest exception cases.
+- @NegativeTest - Tests specifically designed to try and break the software
+- @SmokeTest - Tests that can be used as a builds smoke or sanity tests.
+- @NeedsReview - Tests that have not been reviewed or approved.
+- @ReviewedApproved - obvious
+- @wip - Tests that are a work in progress and incomplete
+- @Done - Finished tests that are expected to Pass if executed against an Agent.
+- @AIP10 - Aries Interop Profile version the tests are written for
+- @T01-API10-RFC0160 - Test Unique Identifier - Please use T for the test cases number, the API version number the test is written for, and the RFC number for the Protocol under test. 
+- @WillFail - Tests completed but because of an outstanding bug, they are expected to fail. The test harness looks for this to report on the next tag  
+- @OutstandingBug..###..url - When using @WillFail, this tag much also be used. It is comprised of 3 components, OutstandingBug, the bug number, and the bug url. These must be separated by double periods.
+
+### Defining Protocol Suite Specific Tags
+There will be cases where there will be a need for Protocol specific tags. This will usually reveal itself when there are optional implementations or where implementations can diverge into 2 or more options. Tests will need to be tagged with the least common option, where no tag means the other option. For example in the connection protocol there are specific tests that exercise the behaviour of the protocol under a Multi Use Invite and a Single Use Invite. The tag @MultiUseInvite is used to differentiate the two, and by default it is expected that MultiUseInvite is the least common option.
+
+Currently Existing Connection Protocol Tags
+- @MultiUseInvite - Test utilizes a multi-use invite. Not using this tag and the test expects the invite to be single use.
+- @SingleTryOnException
+- @RetryableOnException
+
+Defining specific tags should be discussed with the Aries Protocol test community. 
+
+## Defining Backchannel Commands
+
+While defining tests it will become evident that your backchannel, the interface between the tests and your Agent, will need to support certain commands. These are to be defined in the [aries-agent-test-harness/aries-backchannels/backchannel_operations.csv](https://github.com/bcgov/aries-agent-test-harness/blob/master/aries-backchannels/backchannel_operations.csv)
```

Comment:
This csv file is generated from a Google sheet:   https://bit.ly/AriesTestHarnessScenarios